File: app/src/main/java/com/example/quit/Adapter/badgeAdapter.java
package com.example.quit.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.quit.Domain.badgeDomain;
import com.example.quit.R;

import java.util.ArrayList;

public class badgeAdapter extends RecyclerView.Adapter<badgeAdapter.Viewholder> {
    ArrayList<badgeDomain> items;
    Context context;

    public badgeAdapter(ArrayList<badgeDomain> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public badgeAdapter.Viewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View inflator = LayoutInflater.from(parent.getContext()).inflate(R.layout.badge_list,parent,false);
        context = parent.getContext();
        return new Viewholder(inflator);
    }

    @Override
    public void onBindViewHolder(@NonNull badgeAdapter.Viewholder holder, int position) {
        holder.gold_number.setText(items.get(position).getGoldNumber());
        holder.silver_number.setText(items.get(position).getSilverNumber());
        holder.copper_number.setText(items.get(position).getCopperNumber());
        holder.describe.setText(items.get(position).getTitle());
        holder.describe.setText(items.get(position).getDescribe());



        holder.layout.setBackgroundResource(R.drawable.badge_background1);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public class Viewholder extends RecyclerView.ViewHolder{
        TextView gold_number, silver_number, copper_number, title, describe;
        ConstraintLayout layout;
        public Viewholder(@NonNull View itemView) {
            super(itemView);
            gold_number = itemView.findViewById(R.id.gold_number_1);
            silver_number = itemView.findViewById(R.id.silver_number_1);
            copper_number = itemView.findViewById(R.id.copper_number_1);
            title = itemView.findViewById(R.id.title);
            describe = itemView.findViewById(R.id.describe_1);

            layout = itemView.findViewById(R.id.badge_constraintlayout);
        }
    }
}

File: app/src/main/java/com/example/quit/Adapter/badgeAdapter.kt
package com.example.quit.Adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.quit.Domain.badgeDomain
import com.example.quit.R

class badgeAdapter : RecyclerView.Adapter<badgeAdapter.ViewHolder>() {

    //獎章清單??
    private val badgeList = ArrayList<badgeDomain>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        //實現在哪裡
        val inflator = LayoutInflater.from(parent.getContext()).inflate(
            R.layout.badge_list,parent,false)
        return ViewHolder(inflator)
    }

    override fun getItemCount(): Int {
        return badgeList.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val currentitem = badgeList[position]

        holder.title.text = currentitem.title
        holder.describe.text = currentitem.describe
        holder.gold_number.text = currentitem.goldNumber
        holder.silver_number.text = currentitem.silverNumber
        holder.copper_number.text = currentitem.copperNumber
    }

    fun updateBadgeList(badgeList: List<badgeDomain>){
        this.badgeList.clear()
        this.badgeList.addAll(badgeList)
        notifyDataSetChanged()
    }

    //連結示圖中要顯示的地方
    class ViewHolder(itemView : View) : RecyclerView.ViewHolder(itemView){
        val gold_number : TextView = itemView.findViewById(R.id.gold_number_1);
        val silver_number : TextView = itemView.findViewById(R.id.silver_number_1);
        val copper_number : TextView = itemView.findViewById(R.id.copper_number_1);
        val title : TextView = itemView.findViewById(R.id.title);
        val describe : TextView = itemView.findViewById(R.id.describe_1);
    }
}